### PR TITLE
Made hw1

### PR DESCRIPTION
**Pull request by smellofnapalm**
Waiting for response.

### CODE REVIEW

File: src/test/java/edu/hw1/Task1Test.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package edu.hw1;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import edu.hw1.Task1;
+
+public class Task1Test {
```
Comment (vasilmik) [inline src/test/java/edu/hw1/Task1Test.java:8]: Необходимо проверить на corner case, когда напишете проверку поймете что алгоритм неверный

File: src/test/java/edu/hw1/Task2Test.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package edu.hw1;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+public class Task2Test {
```
Comment (vasilmik) [inline src/test/java/edu/hw1/Task2Test.java:7]: Необходимо проверить на corner case, когда напишете проверку поймете что алгоритм в некотором случае неверный

File: src/main/java/edu/hw1/Task5.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
+package edu.hw1;
+
+import static java.lang.Math.abs;
+
+public final class Task5 {
+    public static boolean isPalindrome(String x) {
```
Comment (vasilmik) [inline src/main/java/edu/hw1/Task5.java]: не делайте методы public если не вызываете его вне класса

File: src/main/java/edu/hw1/Task5.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            }
+            if (isPalindrome(x)) {
+                return true;
+            }
+            x = makeStep(x);
+        }
+        return false;
+    }
+
+    private Task5() {
```
Comment (vasilmik) [inline src/main/java/edu/hw1/Task5.java:53]: Это нужно удалить, не оставляйте неиспользуемый код

File: src/main/java/edu/hw1/Task6.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        final int lowerBound = 1000;
+        final int upperBound = 10000;
+        final int digitsEqual = 1111;
+        int n = x;
+        if (n <= lowerBound || n >= upperBound || n % digitsEqual == 0) {
+            return -1;
+        }
+        int ans = 0;
+        while (n != stablePoint) {
+            n = kaprekar(n);
```
Comment (vasilmik) [inline src/main/java/edu/hw1/Task6.java]: Попробуйте решить через рекурсию

File: src/main/java/edu/hw1/Task6.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+            res += ten * a[i];
+            ten *= base;
+        }
+        return res;
+    }
+
+    public static int kaprekar(int n) {
+        final int base = 10;
+        final int sz = 4;
+        // Создадим массив из 4 цифр
```
Comment (vasilmik) [inline src/main/java/edu/hw1/Task6.java:22]: Думаю что в этой задаче можно сортировать через строку, в любом случае лучше называть методы согласно их бизнес логике например get...

File: src/main/java/edu/hw1/Task7.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+            res += two * a[i];
+            two *= 2;
+        }
+        return res;
+    }
+
+    public static int[] shiftLeft(int[] a, int shift) {
+        int length = a.length;
+        int[] b = new int[length];
+        System.arraycopy(a, shift, b, 0, length - shift);
```
Comment (vasilmik) [inline src/main/java/edu/hw1/Task7.java:33]: если интересно попробуйте решить задачу без выделения дополнительной памяти, но это не обязательно

File: src/main/java/edu/hw1/Task8.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        if (a == null) {
+            return true;
+        }
+        // Хочу, чтобы программа работала с произвольным размером поля, не только 8x8
+        int n = a.length;
+        int m = 0;
+        if (n >= 1) {
+            m = a[0].length;
+        }
+        var shifts = new Vector<Pair<Integer, Integer>>();
```
Comment (vasilmik) [inline src/main/java/edu/hw1/Task8.java]: Убедитесь что вам действительно нужен класс Vector для этой. задачи

File: src/main/java/edu/hw1/Task5.java (hunk 4 above)
Comment (smellofnapalm) [reply src/main/java/edu/hw1/Task5.java:53]: Если вы про приватный конструктор, то Maven ругается на его отсутствие, я бы сам конечно его не делал.

File: src/test/java/edu/hw1/Task1Test.java (hunk 1 above)
Comment (smellofnapalm) [reply src/test/java/edu/hw1/Task1Test.java:8]: Исправил

File: src/test/java/edu/hw1/Task2Test.java (hunk 2 above)
Comment (smellofnapalm) [reply src/test/java/edu/hw1/Task2Test.java:7]: Исправил

File: src/main/java/edu/hw1/Task5.java (hunk 3 above)
Comment (smellofnapalm) [reply src/main/java/edu/hw1/Task5.java]: Исправил

File: src/main/java/edu/hw1/Task6.java (hunk 5 above)
Comment (smellofnapalm) [reply src/main/java/edu/hw1/Task6.java]: Сделал

File: src/main/java/edu/hw1/Task6.java (hunk 6 above)
Comment (smellofnapalm) [reply src/main/java/edu/hw1/Task6.java:22]: Метод переименовал, в остальном оставил так, строки же неизменяемы, они сортируются приведением к char[], или я не прав?

File: src/main/java/edu/hw1/Task7.java (hunk 7 above)
Comment (smellofnapalm) [reply src/main/java/edu/hw1/Task7.java:33]: Можно при помощи битовых операций и масок сделать все то же самое, я понимаю как, но писать сейчас лень)

File: src/main/java/edu/hw1/Task8.java (hunk 8 above)
Comment (smellofnapalm) [reply src/main/java/edu/hw1/Task8.java]: Сделал int[][] здесь

File: src/main/java/edu/hw1/Task5.java (hunk 4 above)
Comment (smellofnapalm) [reply src/main/java/edu/hw1/Task5.java:53]: Utility classes should not have a public or default constructor. [HideUtilityClassConstructor]
